pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -180,7 +180,7 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
       }
     } else {
       // Raw index
-      _forwardIndex = loadRawForwardIndex(fwdIndexBuffer, metadata.getDataType());
+      _forwardIndex = loadRawForwardIndex(fwdIndexBuffer, metadata.getDataType().getStoredType());
```

Comment:
Can you fix this in `loadRawForwardIndex()`? We might still want to pass in the original data type

Reply:
ok